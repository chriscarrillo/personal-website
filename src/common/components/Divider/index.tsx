import styles from 'common/components/Divider/Divider.module.scss'
import React from 'react'
import {BsCode} from 'react-icons/bs'

interface Props {
  color: 'primary' | 'secondary' | 'dark' | 'light'
}

/**
 * Common divider used across components.
 * @param root0 Divider props
 * @param root0.color Color of the divider
 * @return React component
 */
export const Divider: React.FC<Props> = ({color}: Props) => (
  <div className="d-flex w-100 align-items-center justify-content-center">
    <div className={`bg-${color} mb-2 mr-1 ${styles.divider}`} />
    <BsCode className={`text-${color} mb-2`} size={40} />
    <div className={`bg-${color} mb-2 ml-1 ${styles.divider}`} />
  </div>
)
