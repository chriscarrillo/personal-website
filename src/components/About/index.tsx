import React from 'react'
import {Col, Container, Row} from 'react-bootstrap'

/**
 * About section.
 * @return React components
 */
export const About: React.FC = () => (
  <section className="pt-2">
    <Container fluid className="d-flex flex-column align-items-center justify-content-center">
      <h1 className="text-dark text-uppercase family-semi pb-2 mb-0">About</h1>
      <Row>
        <Col md={6} sm={12}>
          <span>Picture will be here.</span>
        </Col>
        <Col md={6} sm={12}>
          <p>I&apos;m Chris, a software engineer based in Dallas, Texas.</p>
          <p>
            I am passionate about the ins and outs of applications that you see on the web. That
            includes front-end libraries and frameworks, back-end API&apos;s, and relational and
            non-relational databases.
          </p>
          <p>
            After graduating from California Baptist University with a Bachelor of Science in
            Software Engineering, I moved to Dallas and started my career as a software engineer,
            creating intentional and meaningful applications.
          </p>
          <p>Just a few of my favorite technologies:</p>
          <Row>
            <Col className="pl-0" md={4} sm={12}>
              <ul>
                <li>React</li>
                <li>Angular</li>
                <li>Gatsby</li>
              </ul>
            </Col>
            <Col className="pl-0" md={4} sm={12}>
              <ul>
                <li>Commerce.js</li>
                <li>C#</li>
                <li>Java</li>
              </ul>
            </Col>
            <Col className="pl-0" md={4} sm={12}>
              <ul>
                <li>SQL Server</li>
                <li>Firebase</li>
                <li>Azure</li>
              </ul>
            </Col>
          </Row>
        </Col>
      </Row>
    </Container>
  </section>
)
