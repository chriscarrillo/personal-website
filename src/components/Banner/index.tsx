import avatar from 'assets/images/avatar.svg'
import styles from 'components/Banner/Banner.module.scss'
import React from 'react'
import {Element} from 'react-scroll'

/**
 * Banner for home page.
 * @return React component
 */
export const Banner: React.FC = () => (
  <Element name="home">
    <div
      className={`
        ${styles.banner}
        bg-dark
        w-100
        d-flex
        flex-column
        justify-content-center
        align-items-center
        text-center
        text-light`}
    >
      <img alt="Avatar" className="pt-2" src={avatar} />
      <h1 className="pt-2 pb-1 mb-0 family-semi">Chris Carrillo</h1>
      <h2 className="h6 pb-2">Software Engineer - Coffee Lover - Enneagram Enthusiast</h2>
    </div>
  </Element>
)
