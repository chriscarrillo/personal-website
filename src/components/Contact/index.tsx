import styles from 'components/Contact/Contact.module.scss'
import {Form, Formik, FormikProps} from 'formik'
import {ContactFormValues} from 'models/FormValues'
import React from 'react'
import {Form as BootstrapForm, Button, Container} from 'react-bootstrap'
import * as Yup from 'yup'

const initialFormValues: ContactFormValues = {
  email: '',
  message: '',
  name: '',
  subject: '',
}

const ContactFormSchema = Yup.object().shape({
  email: Yup.string().email().required('An email is required'),
  message: Yup.string().required('A message is required'),
  name: Yup.string().required('A name is required'),
  subject: Yup.string().required('A subject is required'),
})

/**
 * Contact section.
 * @return React components
 */
export const Contact: React.FC = () => {
  /**
   * Submit the form.
   */
  function handleSubmit() {
    console.log('hey')
  }
  return (
    <section className={`py-4 text-light ${styles.contact}`}>
      <Container fluid className="d-flex flex-column align-items-center justify-content-center">
        <h1 className="text-uppercase family-semi pb-2 mb-0">Contact</h1>
        <Formik
          validateOnBlur
          initialValues={initialFormValues}
          validationSchema={ContactFormSchema}
          onSubmit={handleSubmit}
        >
          {(props: FormikProps<ContactFormValues>) => (
            <Form className={styles.form}>
              <BootstrapForm.Group controlId="name">
                <BootstrapForm.Control
                  placeholder="Name"
                  type="text"
                  value={props.values.name}
                  onBlur={props.handleBlur}
                  onChange={props.handleChange}
                />
                {props.touched.name !== undefined && props.errors.name && (
                  <BootstrapForm.Text className="text-danger">
                    {props.errors.name}
                  </BootstrapForm.Text>
                )}
              </BootstrapForm.Group>

              <BootstrapForm.Group controlId="email">
                <BootstrapForm.Control
                  placeholder="Email"
                  type="email"
                  value={props.values.email}
                  onBlur={props.handleBlur}
                  onChange={props.handleChange}
                />
                {props.touched.email !== undefined && props.errors.email && (
                  <BootstrapForm.Text className="text-danger">
                    {props.errors.email}
                  </BootstrapForm.Text>
                )}
              </BootstrapForm.Group>

              <BootstrapForm.Group controlId="subject">
                <BootstrapForm.Control
                  placeholder="Subject"
                  type="text"
                  value={props.values.subject}
                  onBlur={props.handleBlur}
                  onChange={props.handleChange}
                />
                {props.touched.subject !== undefined && props.errors.subject && (
                  <BootstrapForm.Text className="text-danger">
                    {props.errors.subject}
                  </BootstrapForm.Text>
                )}
              </BootstrapForm.Group>

              <BootstrapForm.Group controlId="message">
                <BootstrapForm.Control
                  as="textarea"
                  placeholder="Message"
                  rows={4}
                  value={props.values.message}
                  onBlur={props.handleBlur}
                  onChange={props.handleChange}
                />
                {props.touched.message !== undefined && props.errors.message && (
                  <BootstrapForm.Text className="text-danger">
                    {props.errors.message}
                  </BootstrapForm.Text>
                )}
              </BootstrapForm.Group>
              <div className="d-flex justify-content-center py-1">
                <Button className="mr-2" type="submit">
                  Send
                </Button>
                <Button type="reset" variant="secondary">
                  Reset
                </Button>
              </div>
            </Form>
          )}
        </Formik>
      </Container>
    </section>
  )
}
