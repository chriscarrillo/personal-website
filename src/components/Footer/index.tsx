import styles from 'components/Footer/Footer.module.scss'
import React from 'react'
import {Col, Container, Row} from 'react-bootstrap'
import {FaGithub, FaInstagram, FaLinkedin} from 'react-icons/fa'

/**
 * Main footer.
 * @return React component
 */
export const Footer: React.FC = () => (
  <footer className={`${styles.footer} bg-dark d-flex text-light`}>
    <Container
      className="
      d-flex flex-column align-items-center justify-content-center
    "
    >
      <Row noGutters>
        <Col className="mx-1">
          <a
            className="text-light"
            href="https://www.github.com/chriscarrillo/"
            rel="noreferrer"
            target="_blank"
          >
            <FaGithub className="social-link" size={32}>
              GitHub
            </FaGithub>
          </a>
        </Col>
        <Col className="mx-1">
          <a
            className="text-light"
            href="https://www.linkedin.com/in/chriscarrillo/"
            rel="noreferrer"
            target="_blank"
          >
            <FaLinkedin className="social-link" size={32}>
              LinkedIn
            </FaLinkedin>
          </a>
        </Col>
        <Col className="mx-1">
          <a
            className="text-light"
            href="https://www.instagram.com/chriscarrillo7/"
            rel="noreferrer"
            target="_blank"
          >
            <FaInstagram className="social-link" size={32}>
              Instagram
            </FaInstagram>
          </a>
        </Col>
      </Row>
      <Row noGutters>
        <Col>
          <p className="mb-0 pt-1">
            <span className="text-primary">Chris Carrillo</span> &copy; 2020
          </p>
        </Col>
      </Row>
    </Container>
  </footer>
)
