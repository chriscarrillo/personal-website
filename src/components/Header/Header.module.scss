@use 'src/theme/variables';

.navbar {
  border-bottom: 2px solid variables.$secondary;
  height: 60px;
}

.navLink {
  &:hover {
    cursor: pointer;
  }
}
