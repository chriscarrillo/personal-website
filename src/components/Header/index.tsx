import styles from 'components/Header/Header.module.scss'
import React from 'react'
import {Nav, Navbar} from 'react-bootstrap'
import {NavLink} from 'react-router-dom'
import {Link} from 'react-scroll'

/**
 * Main header.
 * @return React component
 */
export const Header: React.FC = () => (
  <header className="text-uppercase">
    <Navbar
      collapseOnSelect
      bg="dark"
      className={styles.navbar}
      expand="md"
      fixed="top"
      variant="dark"
    >
      <Navbar.Brand exact as={NavLink} to="/">
        Chris Carrillo
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav className="ml-auto">
          <Nav.Link
            smooth
            spy
            activeClass="text-primary"
            as={Link}
            className={styles.navLink}
            eventKey={1}
            offset={-60}
            to="home"
          >
            Home
          </Nav.Link>
          <Nav.Link
            smooth
            spy
            activeClass="text-primary"
            as={Link}
            className={styles.navLink}
            eventKey={2}
            offset={-60}
            to="about"
          >
            About
          </Nav.Link>
          <Nav.Link
            smooth
            spy
            activeClass="text-primary"
            as={Link}
            className={styles.navLink}
            eventKey={3}
            offset={-60}
            to="projects"
          >
            Projects
          </Nav.Link>
          <Nav.Link
            smooth
            spy
            activeClass="text-primary"
            as={Link}
            className={styles.navLink}
            eventKey={4}
            offset={-60}
            to="contact"
          >
            Contact
          </Nav.Link>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  </header>
)
