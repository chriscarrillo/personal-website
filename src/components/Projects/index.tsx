import klawWeb from 'assets/images/klaw-web.png'
import {Project} from 'models'
import React from 'react'
import {Card, CardDeck, Col, Container, Row} from 'react-bootstrap'
import {FaExternalLinkAlt, FaGithub} from 'react-icons/fa'
import {MdDeveloperMode} from 'react-icons/md'

const projects: Project[] = [
  {
    description: 'A personal website that showcases my projects and shares more info about me.',
    image: '',
    link: '',
    name: 'Portfolio',
    repositoryLink: 'https://github.com/chriscarrillo/personal-website',
    technologies: ['React', 'TypeScript'],
  },
  {
    description: `
      A web application to create meal plans and add meals for those corresponding plans.`,
    image: '',
    link: '',
    name: 'Swole Your Rolls',
    repositoryLink: 'https://github.com/chriscarrillo/swole-your-rolls',
    technologies: ['React', 'TypeScript', 'Firebase'],
  },
  {
    description: `
      An e-commerce, business website for a life and ennegram coach in which customers can buy
      typing sessions and learn more about the enneagram.`,
    image: '',
    link: '',
    name: 'Kenna Ledbetter Coaching',
    repositoryLink: '',
    technologies: ['React', 'TypeScript', 'Commerce.js'],
  },
  {
    description: `
      An educational website designed to help STEM students and hobbyists learn about
      robotics and programming.`,
    image: klawWeb,
    link: 'https://klaw-web.herokuapp.com/',
    name: 'KLAW-Web',
    repositoryLink: 'https://github.com/chriscarrillo/KLAW-Web',
    technologies: ['Angular', 'JavaScript', 'Blockly', 'Flask', 'Python', 'WebGL'],
  },
  {
    description: 'Android GUI Calculator for frequently used calculations.',
    image: '',
    link: '',
    name: 'Napkin',
    repositoryLink: 'https://github.com/chriscarrillo/Napkin',
    technologies: ['Java'],
  },
  {
    description: `
      A recipe finder website that allows users to quickly and easily find recipes that fit into
      their dietary restrictions and/or what they have available to them.`,
    image: '',
    link: '',
    name: 'Reciplease',
    repositoryLink: 'https://github.com/chriscarrillo/Reciplease',
    technologies: ['PHP', 'Python'],
  },
]

/**
 * Projects section.
 * @return React components
 */
export const Projects: React.FC = () => {
  const projectsRow = projects.map(project => {
    const projectImage =
      project.image.length === 0 ? (
        <div className="p-2 text-center text-light bg-dark">
          <Card.Title>
            <MdDeveloperMode size={40} />
          </Card.Title>
          <Card.Text>This project is still in development.</Card.Text>
        </div>
      ) : (
        <Card.Img alt={project.name} src={project.image} />
      )
    const technologies = project.technologies.map((technology, index) => (
      <span key={technology}>
        {technology}
        {index < project.technologies.length - 1 ? ' â€” ' : ''}
      </span>
    ))
    const repoLink = project.repositoryLink !== undefined && project.repositoryLink.length > 0 && (
      <a className="text-dark pr-2" href={project.repositoryLink} rel="noreferrer" target="_blank">
        <FaGithub className="social-link" size={26}>
          GitHub
        </FaGithub>
      </a>
    )
    const projectLink = project.link !== undefined && project.link.length > 0 && (
      <a
        className="text-dark"
        href={project.link}
        rel="noreferrer"
        target="_blank"
        title="See it live!"
      >
        <FaExternalLinkAlt className="social-link" size={26}>
          {project.name}
        </FaExternalLinkAlt>
      </a>
    )

    return (
      <Col key={project.name} as={CardDeck} className="py-2" md={4} sm={12}>
        <Card>
          <Card.Header className="d-flex justify-content-between">
            <span className="h4 family-semi">{project.name}</span>
            {projectLink}
          </Card.Header>
          {projectImage}
          <Card.Body>
            <Card.Text>{project.description}</Card.Text>
          </Card.Body>
          <Card.Footer>
            <span className="d-block family-semi">Technologies:</span>
            {technologies}
            <div className="pt-1">
              {repoLink}
              {projectLink}
            </div>
          </Card.Footer>
        </Card>
      </Col>
    )
  })

  return (
    <section className="bg-light py-4">
      <Container fluid className="d-flex flex-column align-items-center justify-content-center">
        <h1 className="text-dark text-uppercase family-semi pb-2 mb-0">Projects</h1>
        <Row className="justify-content-center">{projectsRow}</Row>
      </Container>
    </section>
  )
}
