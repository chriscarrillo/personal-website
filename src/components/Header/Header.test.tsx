import {describe, expect, it} from '@jest/globals'
import {render, screen} from '@testing-library/react'
import {Header} from 'components/Header'
import React from 'react'
import renderer from 'react-test-renderer'

describe('<Header />', () => {
  it('should render as expected', () => {
    const tree = renderer.create(<Header />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should show brand', async () => {
    const {findByText} = render(<Header />)
    await expect(findByText('Chris Carrillo')).toBeTruthy()
  })

  it('should show all links', async () => {
    render(<Header />)
    const links = await screen.findAllByTestId('navLink')
    expect(links).toBeTruthy()
    expect(links).toHaveLength(4)
  })
})
