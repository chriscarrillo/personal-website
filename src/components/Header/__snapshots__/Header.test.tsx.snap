// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render as expected 1`] = `
<header
  className="text-uppercase"
>
  <nav
    className="navbar navbar navbar-expand-md navbar-dark bg-dark fixed-top"
  >
    <span
      className="navbar-brand"
    >
      Chris Carrillo
    </span>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="navLink nav-link"
          data-rb-event-key="1"
          data-testid="navLink"
          disabled={false}
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="navLink nav-link"
          data-rb-event-key="2"
          data-testid="navLink"
          disabled={false}
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="navLink nav-link"
          data-rb-event-key="3"
          data-testid="navLink"
          disabled={false}
          onClick={[Function]}
        >
          Projects
        </a>
        <a
          className="navLink nav-link"
          data-rb-event-key="4"
          data-testid="navLink"
          disabled={false}
          onClick={[Function]}
        >
          Contact
        </a>
      </div>
    </div>
  </nav>
</header>
`;
